tools/clang/tools/dxcompiler/dxillib.cpp

```diff
@@ -25,15 +25,17 @@ static llvm::sys::Mutex *cs = nullptr;
 // This function is to prevent multiple attempts to load dxil.dll 
 HRESULT DxilLibInitialize() {
   cs = new llvm::sys::Mutex;
+#if LLVM_ON_WIN32
+  cs->lock();
+  g_DllLibResult = g_DllSupport.InitializeForDll(L"dxil.dll", "DxcCreateInstance");
+  cs->unlock();
+#endif
   return S_OK;
 }
 
 HRESULT DxilLibCleanup(DxilLibCleanUpType type) {
   HRESULT hr = S_OK;
-  if (type == DxilLibCleanUpType::ProcessTermination) {
```

Comment:
Why did you remove this case?
shouldn't something like `m_dll = DynamicLibrary();` happen in this case too?

Reply:
It could, but it doesn't free any resources. The open Dynamic libraries persist until they are all closed by llvm shutdown. I suppose I could do that too keep m_dll from being used again. 

Is there a way to make the powershell script more verbose? It prints a lot when I run it locally, but it prints almost nothing in the automatic appveyor output. I'm wondering it there's an equivalent to echo on somewhere.

Reply:
I'm sure there's a better way to do it than this, but: temporarily, you can change the appveyor.yml file to run `utils\hct\hcttest -rel` at the `build_script` stage to see the test outputs.

Reply:
After looking at it closer, I don't think it's the hcttest run that's failing because the powershell script appveyor_test.ps1 prints the last notices and then returns the result at the end. I think the direct invocation of te is what's failing. I could try to swap that in though. Barring that, I can try tweaking the change in a sort of binary search removing changes to try to get a  better handle on what's going on, but I hope it doesn't come to bouncing off appveyor like that.